contracts/competition/Competition.sol

```diff
@@ -0,0 +1,169 @@
+        Competition storage competition = competitions[competitionIndex];
+
+        require(referralStorage.codeOwners(referralCode) != address(0), "Referral code does not exist.");
+        require(competition.teamNames[name] == false, "Team name already registered.");
+
+        Team storage team = competition.teams[msg.sender];
+        team.leader = msg.sender;
+        team.name = name;
+        team.referralCode = referralCode;
+        team.members.push(msg.sender);
+
+        competition.teamNames[name] = true;
+        competition.memberTeams[msg.sender] = msg.sender;
+
+        emit TeamCreated(competitionIndex, msg.sender, name, referralCode);
+    }
+
+    function createJoinRequest(uint competitionIndex, address leaderAddress) external registrationIsOpen(competitionIndex) isNotMember(competitionIndex) {
+        Competition storage competition = competitions[competitionIndex];
+
+        require(competition.memberTeams[msg.sender] == address(0), "You can't join multiple teams.");
+        require(competition.teams[leaderAddress].leader != address(0), "The team does not exist.");
+
+        competition.joinRequests[msg.sender] = leaderAddress;
+
+        emit JoinRequestCreated(competitionIndex, msg.sender, leaderAddress);
+    }
+
+    function approveJoinRequest(uint competitionIndex, address memberAddress) external registrationIsOpen(competitionIndex) {
+        Competition storage competition = competitions[competitionIndex];
+
+        require(competition.joinRequests[memberAddress] == msg.sender, "This member did not apply.");
+        require(competition.memberTeams[memberAddress] == address(0), "This member already joined a team.");
+        require(competition.teams[msg.sender].members.length < competition.maxTeamSize, "Team is full.");
+
+        // referralStorage.setTraderReferralCode(memberAddress, teams[msg.sender].referral);
+        competition.teams[msg.sender].members.push(memberAddress);
+        competition.memberTeams[memberAddress] = msg.sender;
+        competition.joinRequests[memberAddress] = address(0);
+
+        emit JoinRequestApproved(competitionIndex, memberAddress, msg.sender);
+    }
+
+    function cancelJoinRequest(uint competitionIndex) external registrationIsOpen(competitionIndex) {
+        competitions[competitionIndex].joinRequests[msg.sender] = address(0);
+        emit JoinRequestCanceled(competitionIndex, msg.sender);
+    }
+
+    function removeMember(uint competitionIndex, address memberAddress) external registrationIsOpen(competitionIndex) {
+        Competition storage competition = competitions[competitionIndex];
+
+        require(competition.memberTeams[memberAddress] == msg.sender, "This member is not in your team");
+
+        for (uint i = 0; i < competition.teams[msg.sender].members.length; i++) {
+            if (competition.teams[msg.sender].members[i] == memberAddress) {
+                delete competition.teams[msg.sender].members[i];
```

Comment:
in this case array length will stay the same. and maxTeamSize check might not work correctly